skglm/tests/test_estimators.py

```diff
@@ -97,27 +97,38 @@ def test_check_estimator(estimator_name):
 # Test if skglm solver returns the coefficients
 @pytest.mark.parametrize("estimator_name", dict_estimators_ours.keys())
 @pytest.mark.parametrize('X', [X, X_sparse])
-def test_estimator(estimator_name, X):
+@pytest.mark.parametrize('fit_intercept', [True, False])
+def test_estimator(estimator_name, X, fit_intercept):
     if estimator_name == "GeneralizedLinearEstimator":
         pytest.skip()
     estimator_sk = dict_estimators_sk[estimator_name]
     estimator_ours = dict_estimators_ours[estimator_name]
+    # TODO This seems a bit unusal, maybe to discuss
```

Comment:
@QB3  this has potentially harmful side effects: it sets the intercept to True on an object defined outside the function, so if we first call the function with fit_intercept=True, we set to `model.fit_intercept` to True, then the second time the function is called, the model still has `clf.fit_intercept` set to True.

Reply:
it can be avoided by cloning the estimator when entering the test instead (sklearn.base.clone)